src/dc_tools.rs

```diff
@@ -181,28 +183,29 @@ pub(crate) fn dc_create_smeared_timestamp(context: &Context) -> i64 {
     let now = time();
     let mut ret = now;
 
-    let ts = *context.last_smeared_timestamp.clone().write().unwrap();
-    if ret <= ts {
-        ret = ts + 1;
-        if ret - now > 5 {
-            ret = now + 5
+    let mut last_smeared_timestamp = context.last_smeared_timestamp.write().unwrap();
+    if ret <= *last_smeared_timestamp {
+        ret = *last_smeared_timestamp + 1;
+        if ret - now > MAX_SECONDS_TO_LEND_FROM_FUTURE {
+            ret = now + MAX_SECONDS_TO_LEND_FROM_FUTURE
         }
     }
 
+    *last_smeared_timestamp = ret;
     ret
 }
 
 pub(crate) fn dc_create_smeared_timestamps(context: &Context, count: usize) -> i64 {
```

Comment:
Would you mind adding docstrings to these functions?  You commit message clarifies a lot, i remember stumbling across these before and not really understanding these, having that explanation would be great.
Also This plurals one returns a single value, which I found confusing so explaining the return type would be nice too.